### PR TITLE
ADD: Support reading NEXRAD Level 2 chunks from real-time S3 bucket

### LINKED ISSUE #330: Support reading NEXRAD Level 2 chunks from real-time S3 bucket
## Summary

Add support for reading NEXRAD Level 2 chunk files from the `unidata-nexrad-level2-chunks` S3 bucket.

## Background

The `unidata-nexrad-level2-chunks` bucket contains real-time NEXRAD Level 2 radar data in chunked format before it is assembled into complete volumes in `noaa-nexrad-level2`.

| Bucket | Purpose | Retention |
|--------|---------|-----------|
| `unidata-nexrad-level2-chunks` | Real-time chunks | Max 24 hours |
| `unidata-nexrad-level2` | Complete volumes | Permanent |

## Chunk Structure

**Path:** `s3://unidata-nexrad-level2-chunks/<SITE>/<VOLUME_NUM>/<YYYYMMDD-HHMMSS-CHUNKNUM-CHUNKTYPE>`

**Chunk types:**
| Type | Description | Has Volume Header |
|------|-------------|-------------------|
| S | Start | Yes |
| I | Intermediate | No |
| E | End | No |

Each chunk is bzip2 compressed and contains ~120 Message Type 31 radial records (~100° of azimuth at one elevation).

## Current Limitation

xradar cannot read individual chunk files, particularly Intermediate (I) and End (E) chunks which lack the volume header.

## Related Work

MetPy encountered the same issue and implemented a fix in v1.0:
- Issue: https://github.com/Unidata/MetPy/issues/1470
- Solution: https://github.com/Unidata/MetPy/issues/1470#issuecomment-676615214

The JavaScript [nexrad-level-2-data](https://github.com/netbymatt/nexrad-level-2-data) library also supports reading and combining chunks.

## Discussion

How should xradar handle reading data from the chunks bucket? Looking for community input on the best approach.

### PR DESCRIPTION
**Pull request by aladinor**
## Summary

- Add support for list/tuple of chunk sources (bytes, file-like, or paths) as input to `open_nexradlevel2_datatree` for streaming NEXRAD Level 2 data from the `unidata-nexrad-level2-chunks` S3 bucket
- Add `incomplete_sweep` parameter (`"drop"`/`"pad"`) to handle partial volumes where not all chunks have arrived yet
- Add notebook example demonstrating real-time chunk streaming from S3
- Add comprehensive offline test suite for chunk list-input and incomplete sweep handling

Closes #330

### Core implementation (`xradar/io/backends/nexrad_level2.py`)
- `_concatenate_chunks()`: Assembles a list of chunk byte objects, file-like objects, or file paths into a single byte stream, with automatic volume header detection and skipping for non-start chunks
- `incomplete_sweeps` property on `NEXRADLevel2File`: Tracks which sweeps were force-closed (incomplete) during parsing
- `incomplete_sweep` parameter in `open_nexradlevel2_datatree`:
  - `"drop"` (default): Excludes incomplete sweeps and emits a warning
  - `"pad"`: Keeps incomplete sweeps and reindexes to a full azimuth grid (360 or 720 azimuths), filling missing rays with NaN

### Fix (`xradar/util.py`)
- Use `.values.copy()` instead of `.values` to prevent modifying read-only arrays during angle reindexing

### Notebook (`examples/notebooks/nexrad_read_chunks.ipynb`)
- Full walkthrough: connecting to S3, downloading chunks, opening full/partial volumes, drop vs pad mode, early streaming with few chunks

### Documentation (`docs/importers.md`, `docs/usage.md`, `docs/history.md`)
- User-facing docs for chunk input and incomplete_sweep parameter with code examples
- Notebook registered in docs toctree
- Changelog entries

### Tests (`tests/io/test_nexrad_level2.py`)
- 21 new tests across 5 test classes covering:
  - `_concatenate_chunks` edge cases (empty list, single element, volume header, short chunks, file-like objects, nonexistent paths, tuple input)
  - Sweep completeness detection (multiple incomplete, all incomplete, missing key defaults)
  - `incomplete_sweep` parameter behavior (drop warns, pad passes incomplete set, late-detection path)
  - Pad-mode reindexing with real data (full azimuth grid, data preservation, NaN percentage)
  - List/tuple input integration with drop/pad modes

## Test plan

- [x] All 75 tests pass (`pytest tests/io/test_nexrad_level2.py`)
- [x] Notebook executes successfully against live S3 data
- [x] Docs build successfully (`cd docs && make html`)

### CODE REVIEW

File: docs/importers.md [hunk 1]
```diff
@@ -125,6 +125,56 @@ With {func}`~xradar.io.backends.nexrad_level2.open_nexradlevel2_datatree`
...
+    partial_bytes, incomplete_sweep="drop"
+)
+
+# Pad mode -- all sweeps, missing rays filled with NaN
+dtree = xd.io.open_nexradlevel2_datatree(
+    partial_bytes, incomplete_sweep="pad"
+)
+```
+
+See the ``nexrad_read_chunks`` notebook for a full walkthrough.
```
Comment (kmuehlbauer) [inline docs/importers.md]: Could we ref-link here?

**Review by kmuehlbauer — COMMENTED** [review]
Two quick comments after skimming through this neat PR. I'll have a closer look over the weekend. Thanks @aladinor!

**Review by kmuehlbauer — APPROVED** [review]
There might be improvements possible for the compressed/uncompressed separation, but we should move along here! Thanks @aladinor!

**Review by kmuehlbauer — APPROVED** [review]
Thanks @aladinor! Great it's now possible to already work with brand new NEXRAD data.
